src/PaginatedData.elm

```diff
@@ -1,353 +1,416 @@
-            [ Just <| func (currentPage + 1) ]
-        else
-            []
-    else
-        []
 
+{-| Get a single value, whether it is local or on a page.
 
+We assume that keys are unique. If not, we will return a value with that key,
+but which one is undefined.
 
--- CRUD
+-}
+get : key -> PaginatedData err key value -> Maybe value
+get key (PaginatedData data) =
+    -- We iterate through the pages looking for the key. If
+    -- necessary, we could optimize this by keeping an index
+    -- of which page each key is on.
+    Dict.foldl
+        (\page data accum ->
+            case accum of
+                Just found ->
+                    -- We don't keep testing once we find one.
+                    accum
 
+                Nothing ->
+                    case data of
+                        Success items ->
+                            EveryDictList.get key items
 
-{-| Get a single value.
--}
-get :
-    identifier
-    -> key
-    -> EveryDict identifier (WebData (PaginatedData key value))
-    -> Maybe value
-get identifier key dict =
-    let
-        existing =
-            EveryDict.get identifier dict
-                |> Maybe.withDefault (RemoteData.Success emptyPaginatedData)
+                        _ ->
+                            Nothing
+        )
+        -- We start by looking in local, and then iterate through
+        -- the pages.
+        (EveryDictList.get key data.local)
+        data.pager
 
-        dataAndPager =
-            existing
-                |> RemoteData.toMaybe
-                |> Maybe.withDefault emptyPaginatedData
-    in
-    EveryDictList.get key dataAndPager.data
 
+{-| Get all values, whether from a page or local.
+
+We assume that keys are unique. If not, we will include one value for each key,
+but it is undefined which value will be used.
+
+The values will be returned in order by page, with local values
+at the end.
 
-{-| Get all values.
 -}
-getAll :
-    identifier
-    -> EveryDict identifier (WebData (PaginatedData key value))
-    -> EveryDictList key value
-getAll identifier dict =
-    let
-        existing =
-            EveryDict.get identifier dict
-                |> Maybe.withDefault (RemoteData.Success emptyPaginatedData)
+getAll : PaginatedData err key value -> EveryDictList key value
+getAll (PaginatedData data) =
+    -- We do a `foldr` so that we can append to the beginning of
+    -- the DictList, which is faster than appending at the end.
+    Dict.foldr
+        (\page data accum ->
+            case data of
+                Success items ->
+                    EveryDictList.append items accum
+
+                _ ->
+                    accum
+        )
+        data.local
+        data.pager
 
-        dataAndPager =
-            existing
-                |> RemoteData.toMaybe
-                |> Maybe.withDefault emptyPaginatedData
-    in
-    dataAndPager.data
 
+{-| A convenience to map over our internal pages.
```

Comment:
`internal` here is a bit confusing - maybe `A convenience to map over our the the items in the pages (i.e. not the "local" ones).`